### PR TITLE
Update `eth-json-rpc-infura`

### PR DESCRIPTION
**Pull request by Gudahtt**
The package `eth-json-rpc-infura@5` has been updated to `@metamask/eth-json-rpc-infura@7`. This update includes TypeScript support, and it drops support for older node.js versions. The exports have also been changed from default to named exports.

See here for a full list of changes: https://github.com/MetaMask/eth-json-rpc-infura/blob/main/CHANGELOG.md#700

## Manual Testing Steps

No functional changes

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [x] "Extension QA Board" label has been applied

### CODE REVIEW

**Review by NiranjanaBinoy — APPROVED** [review]
LGTM

File: lavamoat/browserify/policy-override.json [hunk 1]
```diff
@@ -58,6 +58,11 @@
         "XMLHttpRequest": true
       }
     },
+    "@metamask/controllers>web3": {
+      "globals": {
+        "XMLHttpRequest": true
```
Comment (adonesky1) [inline lavamoat/browserify/policy-override.json:63]: I've been having issues with this flipping to "write". Meant to add the override myself, but still don't understand what this is about... @Gudahtt can you explain?

Comment (Gudahtt) [reply lavamoat/browserify/policy-override.json:63]: I cannot. See this commit message:

>The `web3` package used by `@metamask/controllers` unintentionally
overwrites the `XMLHttpRequest` global, which breaks things. This was
fixed by revoking `web3`'s write access to that global using a policy
override.
>
>Previously this policy override was applied to `web3`, but for some
unknown reason, this update caused that override to no longer apply.

There is a thread in the LavaMoat channel about it. I'd recommend following that thread for further updates. We haven't yet identified the problem.

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Pull request merged**